src/components/withKeyboardState.tsx

```diff
@@ -1,20 +1,25 @@
-import React, {forwardRef, createContext, useEffect, useState} from 'react';
+import React, {ComponentType, createContext, ForwardedRef, forwardRef, ReactNode, useEffect, useState} from 'react';
 import {Keyboard} from 'react-native';
 import PropTypes from 'prop-types';
-import getComponentDisplayName from '../libs/getComponentDisplayName';
+import getComponentDisplayName from "../libs/getComponentDisplayName";
 
-const KeyboardStateContext = createContext(null);
-const keyboardStatePropTypes = {
-    /** Whether or not the keyboard is open */
-    isKeyboardShown: PropTypes.bool.isRequired,
+type KeyboardStateContextValue = {
+    /** Whether the keyboard is open */
+    isKeyboardShown : boolean
 };
 
-const keyboardStateProviderPropTypes = {
+type KeyboardStateProviderProps = {
     /* Actual content wrapped by this component */
-    children: PropTypes.node.isRequired,
+    children: ReactNode
 };
 
-function KeyboardStateProvider(props) {
+// TODO: Remove - left for backwards compatibility with existing components.
```

Comment:
Resolve TODO.

Reply:
So, this TODO is waiting for another PR?

Reply:
Okay. Do we have an issue where we can put it is like a requirement to don't forget?

Reply:
Not in this PR about how about adding lint rule to disallow propTypes definition in TS?
Then temporarily disable lint on this line and after relevant components are migrated, we can remove.